jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/SsoSessionConfigurationManager.kt

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.credentials
+
+import com.intellij.openapi.util.io.FileUtil
+import software.amazon.awssdk.profiles.Profile
+import software.amazon.awssdk.profiles.ProfileFile
+import software.amazon.awssdk.profiles.ProfileFileLocation
+import software.amazon.awssdk.profiles.ProfileProperty.SSO_ACCOUNT_ID
+import software.amazon.awssdk.profiles.ProfileProperty.SSO_REGION
+import software.amazon.awssdk.profiles.ProfileProperty.SSO_ROLE_NAME
+import software.amazon.awssdk.profiles.ProfileProperty.SSO_START_URL
+import software.aws.toolkits.jetbrains.core.credentials.SsoProfileConstants.SSO_SESSION_PROFILE_NAME
+import software.aws.toolkits.jetbrains.core.credentials.profiles.SsoSessionConstants.PROFILE_SSO_SESSION_PROPERTY
+import software.aws.toolkits.jetbrains.core.credentials.profiles.SsoSessionConstants.SSO_REGISTRATION_SCOPES
+import software.aws.toolkits.jetbrains.core.credentials.profiles.SsoSessionConstants.SSO_SESSION_SECTION_NAME
+import java.io.File
+import java.util.Optional
+
+class SsoSessionConfigurationManager {
```

Comment:
as discussed offline, tests will be added separately

Reply:
sure